geoengine/workflow.py

```diff
@@ -59,19 +60,21 @@ class Workflow:
     '''
 
     __workflow_id: WorkflowId
+    __result_descriptor: ResultDescriptor
 
     def __init__(self, workflow_id: WorkflowId) -> None:
         self.__workflow_id = workflow_id
+        self.__result_descriptor = self.__query_result_descriptor()
```

Comment:
wouldn't it be better to take the result descriptor as a parameter and get it in the retrieval in the `register_workflow` method? Otherwise it is not possible to create a `Workflow` object that is not yet registered and every `Workflow()` initialization incurs a web request

Reply:
Currently not, since you shouldn't instantiate this without a metadata query.
There shouldn't be a workflow object that is not yet registered.